datalake/src/main/java/com/sequenceiq/datalake/flow/create/handler/EnvWaitHandler.java

```diff
@@ -56,25 +77,65 @@ protected Selectable doAccept(HandlerEvent<EnvWaitRequest> event) {
         try {
             LOGGER.debug("start polling env for sdx: {}", datalakeId);
             DetailedEnvironmentResponse detailedEnvironmentResponse = environmentService.waitAndGetEnvironment(datalakeId);
+
+            // TODO Extract this logic into a separate flow step - CB-17514
+            sdxClusterRepository.findById(datalakeId)
+                    .ifPresentOrElse(this::scheduleStorageConsumptionCollectionIfNeeded,
+                            () -> {
+                                throw notFound("SDX cluster", datalakeId).get();
+                            });
+
             response = new EnvWaitSuccessEvent(datalakeId, userId, detailedEnvironmentResponse);
             setEnvCreatedStatus(datalakeId);
         } catch (UserBreakException userBreakException) {
             LOGGER.error("Env polling exited before timeout. Cause: ", userBreakException);
             response = new SdxCreateFailedEvent(datalakeId, userId, userBreakException);
         } catch (PollerStoppedException pollerStoppedException) {
-            LOGGER.error("Env poller stopped for sdx: {}", datalakeId, pollerStoppedException);
+            LOGGER.error("Env poller stopped for SDX: {}", datalakeId, pollerStoppedException);
             response = new SdxCreateFailedEvent(datalakeId, userId,
                     new PollerStoppedException("Env wait timed out after " + durationInMinutes + " minutes"));
         } catch (PollerException exception) {
-            LOGGER.error("Env polling failed for sdx: {}", datalakeId, exception);
+            LOGGER.error("Env polling failed for SDX: {}", datalakeId, exception);
             response = new SdxCreateFailedEvent(datalakeId, userId, exception);
         } catch (Exception anotherException) {
-            LOGGER.error("Something wrong happened in sdx creation wait phase", anotherException);
+            LOGGER.error("Something wrong happened in SDX creation wait phase", anotherException);
             response = new SdxCreateFailedEvent(datalakeId, userId, anotherException);
         }
         return response;
     }
 
+    private void scheduleStorageConsumptionCollectionIfNeeded(SdxCluster sdxCluster) {
```

Comment:
I would move all this logic into `ConsumptionService` class.
The same is true for the delete flow in `RdsDeletionHandler` .

Reply:
Makes sense. The original `ConsumptionService` has been kept and renamed to `ConsumptionClientService`, and the above logic got moved to `ConsumptionService`.